packages/modal-core/src/utils/CoreUtil.ts

```diff
@@ -36,10 +36,22 @@ export const CoreUtil = {
     return Array.isArray(data) && data.length > 0
   },
 
+  isTelegram() {
+    return (
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      Boolean((window as any).TelegramWebviewProxy) ||
```

Comment:
Shouldn't we check that window is defined first ?
```suggestion
      typeof window !== "undefined" && Boolean((window as any).TelegramWebviewProxy) ||
```

Reply:
+ Yes, directly accessing window without checking if it’s defined might cause issues on server-side envs, where `window` is not available. This would potentially result in a `ReferenceError`